### ISSUE TITLE
Where is the plugin interface?

### ISSUE THREAD

**Issue by GhostofGoes** [sdispater/poetry#869]
- [x] I have searched the [issues](https://github.com/sdispater/poetry/issues) of this repo and believe that this is not a duplicate.

## Issue

The documentation explains how to extend or change the functionality of Poetry using plugins: https://poetry.eustace.io/docs/plugins/

The example shown demonstrates how to create a class that implements a plugin:
```python
from poetry.plugins import Plugin


class MyPlugin(Plugin):
...
```

However, this does not seem to be implemented. I've attempted to import with no success, and can't find any references to a "Plugin" class anywhere in the codebase.

Short version: are plugins a thing, and if they are, what am I doing wrong?

(PS: Thanks for making such an awesome tool!)

**Comment by GhostofGoes** [comment]
Thank you! I'll close this issue then.
